src/asmcnc/comms/user_settings_manager.py

```diff
@@ -0,0 +1,127 @@
+import json
+import os
+import threading
+
+from kivy.clock import Clock
+from kivy.event import EventDispatcher
+from kivy.properties import BooleanProperty, StringProperty, NumericProperty
+
+from asmcnc import paths
+from asmcnc.comms.logging_system.logging_system import Logger
+
+
+class UserSettingsManager(EventDispatcher):
+    """
+    This class manages user settings e.g. if an open dust shoe should be detected or not.
+    All settings:
+     - will be saved into user_settings.json and loaded on startup.
+     - are defined as kivy properties
+
+    Settings have
+     - a type: str ('str', 'bool', 'float')
+     - a name: str (used for internal usage in code, key of the dict entry)
+     - a title: str (used for title labels e.g. settings tab)
+     - a description: str (used for descriptive labels)
+     - a value: corresponding to their specific type
+
+     This settings manager provides getters and setters. Setters will do a validation check.
+
+     Create new settings:
+     Just add the new setting to the settings dictionary
+
+     Usage:
+
+     DON'T INSTANTIATE YOURSELF! GET IT FROM THE APP!
+     user_settings_manager = App.get_running_app().user_settings_manager
+
+     my_bool_attribute = user_settings_manager.get_value('dust_shoe_detection')
+     user_settings_manager.set_value('dust_shoe_detection', value)
+     my_title_label.text = user_settings_manager.get_title('dust_shoe_detection')
+     my_description_label.text = user_settings_manager.get_description('dust_shoe_detection')
+    """
+    SETTINGS_FILE_PATH = os.path.join(paths.SB_VALUES_PATH, "user_settings.json")
+
+    # local copy of settings (Add new settings here):
+    settings = {'dust_shoe_detection': {'description': 'When activated, the dust shoe needs to be inserted '
```

Comment:
Formatting is a bit awkward to look at. Reformatting with pycharm/black would be nice (could be a good idea for all new code, to keep consistent) https://github.com/psf/black

Reply:
That's only because you probably use side-by-side-view. (I also do mostly) but when you switch to unified I think it is pretty easy to read.

Reply:
![image](https://github.com/YetiTool/easycut-smartbench/assets/151084156/6760cc33-42db-4384-b6fe-95426241658c)

That's how it looks in pycharm directly

Reply:
![image](https://github.com/YetiTool/easycut-smartbench/assets/46889734/12f40a6d-aba8-46ed-bb19-a33e0965db9d)

Black formats like this and I think it's more readable.

Reply:
same as a json file

Reply:
Yeah, that's better, but I indented the wrapped description:

![image](https://github.com/YetiTool/easycut-smartbench/assets/151084156/d3148bf7-788d-4892-8dbc-4c20ac411df1)